app/components/InfoHeader.tsx

```diff
@@ -43,6 +48,19 @@ export function InfoHeader({ relatedPaths }: InfoHeaderProps) {
   const [hovering, setHovering] = useState(false);
   console.warn(selectedInfo);
 
+  const newPath = formattedSelectedInfo.replace(/^#/, "$").replace(/\//g, ".");
+
+  const checkPathExists = () => {
+    const heroPath = new JSONHeroPath(newPath);
+    const node = heroPath.first(json);
+    if (node) return true;
+    return false;
+  };
+
+  const handleClick: React.MouseEventHandler<HTMLButtonElement> = () => {
```

Comment:
I think this should be wrapped in a `useCallback` hook with a dependency array of `[newPath, goToNodeId]`